### PR TITLE
GraphicsDeviceManager fixes

### LINKED ISSUE #539: Render to texture: RenderTargetUsage.PreserveContents ignored
Hey,
OS: Linux
I do not exactly know how this feature works, but in many tutorials, the following can be found.
1:  React to the this event in your Game class:
OnPreparingDeviceSettings(Object sender, PreparingDeviceSettingsEventArgs e)
by doing this in the constructor:
mGraphicsDeviceManager.PreparingDeviceSettings += OnPreparingDeviceSettings;

2: Set e.GraphicsDeviceInformation.PresentationParameters.RenderTargetUsage = RenderTargetUsage.PreserveContents;

When you now change the render target with RenderTargetUsage.PreserveContents as the last parameter, the contents of the old render target will not be lost. This allows for render-to-texture code between other rendering-code without getting a purple screen.
MonoGame does not even call OnPreparingDeviceSettings and therefore ignores RenderTargetUsage.PreserveContents.

### LINKED ISSUE #3114: graphicsdevicemanager.disposed not implemented
http://msdn.microsoft.com/en-us/library/microsoft.xna.framework.graphicsdevicemanager.disposed.aspx

Is not implemented

Its not a major part of our code but we have found that logging Device and DeviceManager events into our debugging logs really can help detect issues when you release on PC - there's so many things that can go wrong that clues like these events firing can help

### LINKED ISSUE #4433: Refactor GraphicsDevice.PresentationParameters
Doing some experimenting against XNA for #4425 I found that our implementation of `GraphicsDevice.PresentationParameters` is wrong.

`GraphicsDevice.PresentationParameters` is only the output presentation values representing the state of the `GraphicsDevice` the last time the presentation was changed.

This means `GraphicsDeviceManager` shouldn't be making changes to it directly.  Instead `GraphicsDeviceManager` should be creating a new `PresentationParameters` object... populating it with the new state... then passing that to `GraphicsDeviceManager` to become the new presentation state.

Some code that currently fails under MonoGame:

```
var oldpp = GraphicsDevice.PresentationParameters;
GraphicsDeviceManager.PreferredBackBufferWidth = 100;
GraphicsDeviceManager.ApplyChanges();
var newpp = GraphicsDevice.PresentationParameters; 
Debug.Assert(!object.ReferenceEquals(oldpp, newpp));
```

### PR DESCRIPTION
**Pull request by Jjagg**
This is not finished yet, but I'm sharing so people that are interested and/or worried can follow progress and advise. This is a start to getting GraphicsDevice and GraphicsDeviceManager events right and handling PresentationParameters the same as in XNA.

I may have done a seperate PR for this, but I also added a simple method to TestGameBase that initializes the game without creating a window. This can speed up a lot of test that are 'visual' but don't need to be. I already fixed a couple that just needed a GraphicsDevice, but not necessarily a window... Actually, most tests don't **really** need a window, right? Note that we can't get around that for XNA tests. Other tests that can benefit are RasterizerState and Viewport among others.

There's more testing to be done and GraphicsDevice events also needs its events fixed, but that might be better to do in a seperate PR. I did do the Disposed event though, because of how easy it was and I couldn't resist :p

This fixes at least (I'll add issues here as I find them)
- fixes #539
- fixes #4433
- fixes #3114

### CODE REVIEW

**Comment by Jjagg** [comment]
I'm gonna try and pull out the test speed up commit and get a seperate PR up for those to keep things clear in this one.

File: Test/Framework/Graphics/GraphicsDeviceTest.cs [hunk 1]
```diff
@@ -14,6 +14,94 @@ namespace MonoGame.Tests.Graphics
...
+
+            Assert.IsTrue(vb.IsContentLost);
+            Assert.IsFalse(vb.IsDisposed);
+
+            Assert.IsTrue(ib.IsContentLost);
+            Assert.IsFalse(ib.IsDisposed);
+
+            Assert.IsTrue(rtc.IsContentLost);
+            Assert.IsFalse(rtc.IsDisposed);
+        }
```
Comment (Jjagg) [inline Test/Framework/Graphics/GraphicsDeviceTest.cs]: It'll take a bit of work to get this in order. Not sure how to best go about this. Maybe an IDynamicGraphicsResource that extend IGraphicsResource. Then we can type check through the GraphicsResource collection when GraphicsDevice is lost. Or maybe keep the dynamic ones in a seperate collection. Either way that's a bit large to fit in this PR, so I'll probably ignore this test later on and put a TODO in it.

Comment (tomspilman) [inline Test/Framework/Graphics/GraphicsDeviceTest.cs]: Yeah ignore this test for now.  I think it can be done without any additional interfaces.

**Comment by Jjagg** [comment]
I'm not sure why there are conflicts, I just rebased this on develop. Looking at this in SourceTree, it seems I messed up somewhere. @tomspilman Can you see what exactly the conflicts are? If it's not easily resolvable I'll pull these commits to a new branch.

**Comment by Jjagg** [comment]
Don't know why the Mac build is failing. It's complaining about opentk missing... Might be related to the conflicts and caused by wrong merges since the original was a pretty old PR

**Comment by Jjagg** [comment]
Woops, I confused this with the backbuffer pr! I'll look into why Mac is giving me a hard time.

**Comment by Jjagg** [comment]
Closing in favor of #5139 because of conflicts here.

**Pull request closed without merging**